specs/error_state/ErrorOutOfGasSLOAD.md

```diff
@@ -0,0 +1,41 @@
+# ErrorOutOfGasSLOAD state
+
+## Procedure
+
+Handle the corresponding out of gas error for opcode `SLOAD`.
+
+### EVM behavior
+
+For the current [go-ethereum] code, the out of gas error may occur for [constant gas] or [dynamic gas].
+
+The [default SLOAD constant gas] is set to `50` and no dynamic gas.
+When [EIP-150] is enabled, [EIP-150 SLOAD constant gas] is set as `200`.
+When [EIP-2929] is enabled, [EIP-2929 SLOAD dynamic gas] function is set.
+
+For this gadget, the gas cost is calculated with [EIP-2929] as:
+```
+gas_cost = GAS_COST_WARM_ACCESS if is_warm_access else COLD_SLOAD_COST
+```
+
+### Constraints
+
+1. Constrain `gas_left < gas_cost`.
+2. Current call must fail.
+3. If it's a root call, it transits to `EndTx`.
```

Comment:
add rw_counter_of_reversion_contraint here .

Reply:
Added `rw_counter_end_of_reversion = rw_counter_end_of_step + reversible_counter` constraint in [99df3b7](https://github.com/scroll-tech/zkevm-specs/pull/57/commits/99df3b7196c297ae547833c55d8f5664fdeaf5bd).